coredocument/coredocument.go

```diff
@@ -304,15 +306,54 @@ func GetExternalCollaborators(selfCentID identity.CentID, doc *coredocumentpb.Co
 	return collabs, nil
 }
 
+// GenerateNewSalts generates salts for new document
+func GenerateNewSalts(document proto.Message, prefix string) (*proofs.Salts, error) {
```

Comment:
I know you're saying that we don't need to look at this file but maybe we can actually skip the GenerateNewSalts method altogether. The salts are modified when generating the tree so this is not necessary.

Reply:
I believe then you refer to the other `GenerateNewSalts` in the below model file. The main reason to keep a method like this, is to have explicitness on when the salts are generated and stored in the models. Otherwise I am not sure where to put that call.